addons/pos_self_order/controllers/kiosk_entry.py

```diff
@@ -0,0 +1,98 @@
+    def pos_self_order_kiosk_start(self, config_id=None, access_token=None):
+        if not config_id or not access_token:
+            raise werkzeug.exceptions.NotFound()
+
+        lang = request.httprequest.cookies.get('frontend_lang')
+        if lang is not None:
+            language = get_lang(request.env, lang)
+            if language:
+                lang_code = request.env['res.lang']._lang_get_code(language.code)
+                request.lang = language
+                request.update_context(lang=lang_code)
+
+        pos_config_sudo = request.env["pos.config"].sudo().search([
+            ("id", "=", config_id),
+            ('access_token', '=', access_token)], limit=1)
+
+        if not access_token or not config_id or not pos_config_sudo:
+            raise werkzeug.exceptions.NotFound()
+
+        company = pos_config_sudo.company_id
+        user = pos_config_sudo.current_session_id.user_id
+        pos_config = reduce_privilege(pos_config_sudo, company, user)
+        session_info = request.env["ir.http"].get_frontend_session_info()
+
+        return request.render(
+            'pos_self_order.kiosk_index',
+            {
+                'session_info': {
+                    **session_info,
+                    'currencies': request.env["ir.http"].get_currencies(),
+                    'pos_self_order_data': {
+                        'access_token': pos_config.access_token,
+                        **pos_config._get_self_order_kiosk_data(),
+                    },
+                }
+            }
+        )
+
+    @http.route(
+        "/kiosk/get-bg-image/<int:pos_config_id>/<access_token>/<image>", methods=["GET"], type="http", auth="public"
+    )
+    def pos_self_order_get_bg_image(self, pos_config_id, access_token, image):
+        if not pos_config_id or not access_token:
+            raise werkzeug.exceptions.NotFound()
+
+        pos_config_sudo = request.env["pos.config"].sudo().search([
+            ("id", "=", pos_config_id),
+            ('access_token', '=', access_token)], limit=1)
+
+        if not pos_config_sudo or not pos_config_sudo.self_order_kiosk:
+            raise werkzeug.exceptions.NotFound()
+
+        if image == 'eat':
+            image_field = 'self_order_kiosk_image_eat'
+        elif image == 'home':
+            image_field = 'self_order_kiosk_image_home'
+        elif image == 'brand':
+            image_field = 'self_order_kiosk_image_brand'
+        else:
+            raise werkzeug.exceptions.NotFound()
+
+        return (
+            request.env["ir.binary"]
+            ._get_image_stream_from(pos_config_sudo, field_name=image_field)
+            .get_response()
+        )
+
+    @http.route(
+        "/kiosk/get-category-image/<int:category_id>",
+        methods=["GET"],
+        type="http",
+        auth="public",
+    )
+    def pos_self_order_get_image(self, category_id: int):
+        category = request.env["pos.category"].sudo().browse(category_id)
```

Comment:
The problem is that sometimes the self / kiosk can be accessed without access_token, in which case it will be "read-only",  the user will be able to view the product list but not to order. 

This is why we can't request access_token for images. 

We can, however, load PoS images directly when the page is loaded.

Reply:
All images are now loaded on page load, with the exception of product images.